File: src/main/java/org/minturtle/careersupport/interview/dto/InterviewTemplateResponse.java
package org.minturtle.careersupport.interview.dto;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import org.minturtle.careersupport.interview.entity.InterviewTemplate;

import java.time.Duration;
import java.time.ZoneId;
import java.time.ZonedDateTime;
import java.util.Objects;

@Getter
@AllArgsConstructor
@Builder
public class InterviewTemplateResponse {

    private String id;
    private String theme;
    private ZonedDateTime createdAt;

    public static InterviewTemplateResponse of(InterviewTemplate interviewTemplate){
        return InterviewTemplateResponse.builder()
                .id(interviewTemplate.getId())
                .theme(interviewTemplate.getTheme())
                .createdAt(ZonedDateTime.ofInstant(interviewTemplate.getCreatedAt(), ZoneId.systemDefault()))
                .build();
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        InterviewTemplateResponse that = (InterviewTemplateResponse) o;
        return Objects.equals(id, that.id) &&
                Objects.equals(theme, that.theme) &&
                Objects.equals(createdAt.toEpochSecond(), that.createdAt.toEpochSecond());
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, theme, createdAt.toEpochSecond());
    }
}

File: src/main/kotlin/org/minturtle/careersupport/interview/dto/InterviewTemplateResponse.kt
package org.minturtle.careersupport.interview.dto

import org.minturtle.careersupport.interview.entity.InterviewTemplate
import java.time.ZoneId
import java.time.ZonedDateTime

data class InterviewTemplateResponse(
    val id: String,
    val theme: String,
    val createdAt: ZonedDateTime
    ) {


    companion object{
        fun of(interviewTemplate: InterviewTemplate): InterviewTemplateResponse {
            return InterviewTemplateResponse(
                id = interviewTemplate.id,
                theme = interviewTemplate.theme,
                createdAt = ZonedDateTime.ofInstant(interviewTemplate.createdAt, ZoneId.systemDefault())
            )
        }
    }
}


